pyiron/atomistics/structure/atoms.py

```diff
@@ -1499,6 +1499,31 @@ def _scalars_to_hex_colors(scalar_field, start=None, end=None, cmap=None):
             rgb2hex(cmap(scalar)[:3]) for scalar in remapped_field
         ]  # The slice gets RGB but leaves alpha
 
+    @staticmethod
+    def _get_rotation_matrix(axis, theta):
```

Comment:
As somebody has to maintain this code in future I am always wondering if we can reduce the lines in the pyiron code base by using established libraries. In this case, can you do the same with https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.html ?

Reply:
scipy.spatial.transform.Rotation.from_euler seems like it would work. Just note that we need to extend the 3x3 rotation-only matrix to a 4x4 rotation + (zero) translation matrix, so

```python
from scipy.spatial.transform import Rotation
```
```python
R = np.eye(4)
R[:3, :3] = Rotation.from_euler(axis, theta).as_matrix()
```

and then flatten it.

Reply:
This one worked:

```
from scipy.spatial.transform import Rotation
c = np.cos(theta/2)
s = np.sin(theta/2)

r = Rotation.from_quat(np.append(c, s*axis))

R = np.eye(4)
R[:3, :3] = r.as_matrix()

view.control.orient(R.ravel().tolist())
```

Reply:
Would it be better to just update` _get_rotation_matrix()` to Sam's method? Because otherwise, this snippet would need to be run in the notebook to obtain the necessary rotation matrix.

Reply:
On second thought, I will use Micheal's suggestion. Works with fewer lines of code.

Reply:
> On second thought, I will use Micheal's suggestion. Works with fewer lines of code.

It didn't work for me.

Reply:
Try this:

```
from scipy.spatial.transform import Rotation

R = np.eye(4)
axis = 'x'
theta = 90  # in degrees
rotation_matrix = Rotation.from_euler(axis, theta, degrees=True).as_matrix()
R[:3, :3] = rotation_matrix

view.control.orient(R.ravel().tolist())
```

Reply:
Sorry I had erased my comment because I saw how you did it. It was my mistake. Thanks!